### ISSUE TITLE
2 new hardware reported for fresh install on 5M Pro

### ISSUE THREAD

**Issue by madhulk81** [prestonbrown/helixscreen#837]
### HelixScreen Version

0.9.37

### Hardware Platform

Other (describe below)

### Screen / Display

ADM5

### Printer Model

FlashForge Adventurer 5M Pro

### Bug Description

Fresh install of 0.9.37 on KlipperMod 0.6. Installed fine, all calibration ran without issue, small test print without problem. Fans etc all mapped properly during startup wizard.

At startup, the screen displays 2 new hardware reported message, and hardware health screen states that:

2 New Issues,

New Hardware discovered:

controller_fan stepper_driver_fan
fan available but not assigned to any role

fan_generic internal_fan
fan available but not assigned to any role

I can add to expected hardware, but not sure what this will do, and if Helixscreen / Klippermod is correctly assigning my fans, and so will they be run properly during a print?

### Steps to Reproduce

Fresh install of Helix on printer, start up printer.

### Logs

```shell

```

**Comment by prestonbrown** [comment]
Thanks for the report! A couple of things going on here, and I'd like your help nailing down which one bit you.

**What we found**

1. The `ad5m_pro` preset already lists both `controller_fan stepper_driver_fan` and `fan_generic internal_fan` in its expected-hardware set, so under normal flow these shouldn't trigger the "new hardware" warning. The fact that they did suggests the preset wasn't applied to your config in the first place.

2. We dug into the wizard code and found a likely culprit: when auto-detection's printer-type confidence is too low (and Klipper Mod strips some of the proprietary FlashForge sensors that our heuristics rely on, so this is plausible), you pick the printer type from a list — but that manual-selection path doesn't currently re-apply the preset's `hardware/expected` list. Net result: only the items the wizard explicitly maps (bed, hotend, part fan, hotend fan, LED) end up in expected, and everything else looks "new" on next boot.

3. Quick recon on the Klipper Mod repo confirms that the KlipperMod printer.cfg is shared between AD5M and AD5M Pro and exposes the full enclosure-fan set on both (`internal_fan`, `external_fan`, `chamber_fan`, `chamber_led`, `controller_fan stepper_driver_fan`). That makes Klipper Mod's hardware footprint distinct from stock and ForgeX, so it likely deserves its own preset variant alongside the existing ZMOD one.

**Could you grab a debug bundle?**

In Settings → About → tap *Generate Debug Bundle*, then post the share code here. I'm specifically looking at:

- Which preset (if any) ended up applied to your printer config
- The full Klipper object list KlipperMod 0.6 exposes (so we can build a proper `ad5m_pro_klippermod` preset and a detection signature)

Safe to "Add to expected hardware" in the meantime — it just suppresses the warning, won't change how those fans run during prints (Klipper drives them via the firmware config regardless).

**Comment by prestonbrown** [comment]
Quick follow-up after digging into the KlipperMod side.

**No KlipperMod-specific preset is needed.** Our existing `ad5m_pro` preset already lists the exact object names KlipperMod 0.6's `printer.base.cfg` exposes:

| Role | KlipperMod object | In `ad5m_pro` preset |
|------|-------------------|----------------------|
| Part fan | `[fan]` | ✅ `fan` |
| Hotend fan | `[heater_fan hotend_fan]` | ✅ `heater_fan hotend_fan` |
| Stepper driver fan | `[controller_fan stepper_driver_fan]` | ✅ in `expected` |
| Internal recirc | `[fan_generic internal_fan]` | ✅ `aux` + `expected` |
| External / exhaust | `[fan_generic external_fan]` | ✅ `exhaust` + `expected` |
| Chamber recirc | `[fan_generic chamber_fan]` | ✅ `chamber` + `expected` |
| Chamber LED | `[led chamber_led]` | ✅ in `expected` |
| Runout sensor | `[filament_switch_sensor runout_sensor]` | ✅ in `filament_sensors` + `expected` |

So the fix that just landed in `81ad7db5` (wizard manual-pick now applies the preset, same as auto-detect does) should fully resolve this for you. Once it's in the next release, fresh installs on KlipperMod 0.6 + AD5M Pro shouldn't see the "new hardware detected" warnings — the preset's expected list will be merged into your config regardless of whether auto-detect's confidence happens to clear the threshold.

A debug bundle would still be useful to confirm the fix actually catches your exact scenario, but it's no longer blocking — feel free to wait for the release and re-test.

**Comment by madhulk81** [comment]
Thanks Preston - stunningly fast as ever. Happy to test in the next release. I'm struggling to upload a debug bundle: going to Settings -> Help and About -> Upload Debug Bundle, then clicking Upload (with or without klipper logs) is currently crashing helix. If you can tell me where to look in the filesystem, I can manually grab a file for you and upload the old fashioned way :)

I'll be honest and say that I've been playing around the the various Mods (ZMOD, ForgeX and Klipper) to see how they all work, and coupling them with Helix and Guppy, again to see what works and what seems to be the most stable in the 5M constraints.

Happy to test out a fresh install, but don't know how to revert the machine to 100% factory fresh without previous mod stuff still hanging around, if that would be helpful?

**Comment by prestonbrown** [comment]
If your machine has been crashing submitting debug report, that isn't very
useful now is it.  When it comes back up, do you get the submit crash
report dialog?

Preston

**Comment by madhulk81** [comment]
No, I’m not getting the crash report tool coming up after a reboot - it did used to do this a while ago when it was crashing due to a dns lookup issue. Happy to send files etc by hand.Also, the new release .38 isn’t showing error/message boxes properly - it’s stretching them to the height of the screen with only the buttons showing, no text or other controls.AlexOn 20 Apr 2026, at 22:10, Preston Brown ***@***.***> wrote:﻿prestonbrown left a comment (prestonbrown/helixscreen#837)
If your machine has been crashing submitting debug report, that isn't very

useful now is it.  When it comes back up, do you get the submit crash

report dialog?



Preston

**Comment by prestonbrown** [comment]
Yes I know about the error toast issue, working on a fix for tonight for
that.  if you have crash files in config/ please do attach them here.

Preston

**Comment by madhulk81** [comment]
[crash_1.txt](https://github.com/user-attachments/files/26957737/crash_1.txt)
[crash_2.txt](https://github.com/user-attachments/files/26957734/crash_2.txt)
[crash_3.txt](https://github.com/user-attachments/files/26957735/crash_3.txt)
[crash_history.json](https://github.com/user-attachments/files/26957736/crash_history.json)

Hope these are helpful - generated by .41 when trying to upload from the Help and About menu.

**Comment by prestonbrown** [comment]
Tracked the upload crash to root cause. The bundle upload was spawning a raw `std::thread([...]).detach()`, which fails at construction with `EAGAIN` under thread exhaustion on the 5M's kernel — the resulting `std::system_error` escapes through LVGL's C event dispatch and aborts with `std::terminate` (same class of bug that bit us on AD5M wizard camera probing in #724). That's also why the crash files log `uptime=0` and an identical fingerprint across .38/.39/.41: the process never got past the `Upload` click.

Fix in 884099731: route the bundle upload through our bounded-worker `HttpExecutor::slow()` lane instead. I also did a full audit pass on the same anti-pattern across the codebase while I was there — migrated the IPP label printer and Klipper config-editor reconnect poller to `HttpExecutor`, and wrapped the remaining one-shot BT/USB/QR-decode detached spawns in `try/catch(std::system_error)` with a graceful toast + error callback so a thread-exhaustion failure surfaces a user-visible error instead of terminating. Updated CLAUDE.md and added a project lesson so this doesn't creep back in.

Wizard manual-pick preset fix (the original "2 new hardware" warning) was already in 81ad7db5 and ships as part of the same cycle, so a fresh install on KlipperMod 0.6 + AD5M Pro should come up clean now.

Both fixes will be in v0.99.43. Thanks for hanging on long enough to grab those crash files — that pinned it down.
